### PR TITLE
Minimal Windows setup

### PR DESCRIPTION
**Pull request by Lamorak**
## Summary

Allows to run desktop app on Windows
- so far assumes hat the first serial com device is the badge because the name is not specific
- scale of uploaded graphic is a bit off and it is grainy

## How It Was Tested

manually

### CODE REVIEW

File: zeapp/badge/src/windowsMain/kotlin/de/berlindroid/zekompanion/BadgeManager.kt [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+actual typealias Environment = Any
+
+actual fun buildBadgeManager(environment: Environment): BadgeManager = object : JvmBadgeManager() {
+    override fun getBadger2040s() = SerialPort.getCommPorts()
+        .filter {
+            println(it.systemPortName)
+            true
+            // return the first serial comm device, can't find it by descriptive name
+//            it.descriptivePortName.contains("Badger")
```
Comment (mariobodemann) [inline zeapp/badge/src/windowsMain/kotlin/de/berlindroid/zekompanion/BadgeManager.kt:14]: I feel that this doesn't filter anything. Maybe it's a leftover debug statement?

Please remove the comments and don't print something, it feels more noisy then we need.

File: zeapp/badge/build.gradle.kts [hunk 2]
```diff
@@ -88,6 +97,18 @@ kotlin {
...
+                val windowsMain by getting {
+                    dependencies {
+                        implementation(libs.jSerialComm)
+                    }
+
+                    sourceSets {
+                        dependsOn(sourceSets.getAt("jvmMain"))
+                    }
+                }
+            }
```
Comment (mariobodemann) [inline zeapp/badge/build.gradle.kts:111]: Lovely!

File: zeapp/desktop/src/main/kotlin/de/berlindroid/zekompanion/desktop/ui/NameBadgeEditor.kt [hunk 3]
```diff
@@ -31,6 +31,8 @@ import androidx.compose.ui.unit.IntSize
 import androidx.compose.ui.unit.TextUnit
 import androidx.compose.ui.unit.dp
 import androidx.compose.ui.unit.sp
+import de.berlindroid.zekompanion.BADGE_HEIGHT
+import de.berlindroid.zekompanion.BADGE_WIDTH
```
Comment (mariobodemann) [inline zeapp/desktop/src/main/kotlin/de/berlindroid/zekompanion/desktop/ui/NameBadgeEditor.kt:35]: I'm not sure if that change is needed though. Shouldn't it be included already?

**Review by mariobodemann — COMMENTED** [review]
Thanks so much for this, we get closer to having all OSes badge'd.

Some minor remarks, but overall looks good, and simple as an addition.

**Comment by mariobodemann** [comment]
What do you mean with the output being grainy? Can you upload an example? (It could just be a side effect from the dithering...)

File: zeapp/badge/src/windowsMain/kotlin/de/berlindroid/zekompanion/BadgeManager.kt (hunk 1 above)
Comment (Lamorak) [reply zeapp/badge/src/windowsMain/kotlin/de/berlindroid/zekompanion/BadgeManager.kt:14]: this is mostly reason why this is still draft, the device is not called Badger on Windows

**Comment by Lamorak** [comment]
This is the result image. Of course there's dithering involved, but the text should render better. Also the scale is off
PXL_20240418_071257317

File: zeapp/desktop/src/main/kotlin/de/berlindroid/zekompanion/desktop/ui/NameBadgeEditor.kt (hunk 3 above)
Comment (Lamorak) [reply zeapp/desktop/src/main/kotlin/de/berlindroid/zekompanion/desktop/ui/NameBadgeEditor.kt:35]: Dunno, refused to build without

File: zeapp/badge/src/windowsMain/kotlin/de/berlindroid/zekompanion/BadgeManager.kt (hunk 1 above)
Comment (Lamorak) [reply zeapp/badge/src/windowsMain/kotlin/de/berlindroid/zekompanion/BadgeManager.kt]: Updated it with the best possible identifier I could find
Screenshot 2024-04-20 141305

**Comment by milosmns** [comment]
@mariobodemann good to go?

**Review by milosmns — APPROVED** [review]

**Pull request merged**